packages/slackctl/README.md

```diff
@@ -0,0 +1,172 @@
+# slackctl
+
+A utility for managing Slack [App](https://api.slack.com/start/overview) configurations as code with helpful integrations to make dev environments easy.
+
+⚠️ Slack doesn't currently support this. To make this project work, we rely heavily on undocumented apis and client tokens. These tokens should be protected very carefully. Use at your own risk. This is not intended to be used for production, though it does make dev a whole lot easier.
+
+# Table of Contents
+
+- [Install](#install)
+- [Updating a Slack App](#updating-a-slack-app)
+- [App Configuration File](#app-configuration-file)
+  - [Setting up a `local` config](#setting-up-a-local-config)
+- [Getting a session token](#getting-a-session-token)
+
+## Install
+
+Install the package globally for usage as a cli.
+
+```bash
+yarn global add @slack-wrench/slackctl
+# or
+npm install -g @slack-wrench/slackctl
+```
+
+## Updating a Slack App
+
+```bash
+slackctl apply [flags]
+```
+
+Update an existing Slack app with a configuration. It will check the existing configuration, and apply it to match the configuration file. If you made any changes using the UI without updating the config, they will be overridden.
+
+If your event subscription url needs changing, that url must be active and able to respond to a [`url_verification` event](https://api.slack.com/events/url_verification). If you're using the ngrok flag, then your local app should be running before trying to apply the config.
+
+If you are using the ngrok flag, the command will wait for the specific localhost port to become active before trying to apply configuration.
+
+You will need a [session token](#getting-a-session-token) to run this command. These tokens are not supported via Slack, use caution to keep them safe.
+
+You can it via the environment variable `SLACK_SESSION_TOKEN`, a [local config](#setting-up-a-local-config), or via a flag (--session-token)
+
+**Flags:**
+
+- `-s,--session-token=''` - A [session token](#getting-a-session-token) used for authentication
+- `-n,--ngrok='3000'` - Start an [ngrok tunnel](https://ngrok.com/) to your localhost. This will also modify your configuration to change urls to the url exposed from ngrok.
+- `-t,--timeout='10'` - Used with the `--ngrok` flag, how long to wait for your localhost app to become active before timing out.
+
+```bash
+slackctl validate
+```
+
```

Comment:
other potentially relevant commands could be `dry-run` and/or `diff` - which would check/report on differences between local and deployed.

Reply:
I do like those, Is a good question re: how to merge this as it won't be 100% accurate for a while me thinks.

Reply:
see #33